### ISSUE TITLE
Replace exposed filesystem paths in <img> tags with data URIs

### ISSUE THREAD

**Issue by elliot-sawyer** [andrewandante/silverstripe-clippy#34]
When running clippy inside of a docker container (or even an nginx/fpm setup), the internal hostname is exposed along with a filesystem path to the image.  I believe this happens when running the GenerateUserGuides task from CLI rather than the browser.  Would you consider a pull request to convert these to data URIs instead?

**Comment by andrewandante** [comment]
Hey @elliot-sawyer ! That sounds like a reasonable request to me, assuming that it continues to work for non-containerised environments. Can you share what the `<img>` tag comes out like, and what you'd expect it to come out like?

On the surface, I'd say that it sounds like there either needs to be a config option (`html_prefix` or similar) or that `Director::absoluteBaseURL()` is bringing through the "wrong" thing and something else needs to be used instead. More than happy to review a PR if it comes in

**Comment by elliot-sawyer** [comment]
When generating a userguide with an image embedded, the output currently looks something like this:  http://3286ae0bc051cms-user-guide/streamInImage?imagePath=/app/app/docs/_resources/app/docs/img/adding-a-page.png.  Some notes:

- the internal hostname is obviously invalid in a public facing context
- missing the trailing slash
- internal pathname to the image exposed 

I'm proposing that instead, the task could base64-encode the image contents to a data-uri like this: 

```
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJgAAABdBAMAAACrybIVAAAAIGNIUk0AAHomAACAhAAA+gAAAIDoAAB1MAAA6mAAADqYAAAXcJy6UTw...snip...MKsNZMkAAAAASUVORK5CYII=" alt="alt text or filename if missing" />
```

```

**Comment by andrewandante** [comment]
Right, I get you. This seems like a good default option, I can't see any downside, other than it not updating in-place should the file change. So my gut feel is to have this behind a config flag, where the default is your proposed solution and the alternative is the existing solution. More than happy to review a PR if you put it forward 😄
